### ISSUE TITLE
nx caching gives error - Daemon process terminated and closed the connection

### ISSUE THREAD

**Issue by vollmerr** [lerna/lerna#3719]
## Current Behavior

Running a lerna command (ex. `lerna run build`) with nx caching gives the following error
```
 Lerna (powered by Nx)   Daemon process terminated and closed the connection
```

## Expected Behavior

Expected to run the task with caching. This worked on lerna 6.6.2 for me, but migrating to 7.0.0 gives this error.

## Steps to Reproduce

using the [getting started example](https://github.com/lerna/getting-started-example) (version 6), update to 7.0.0 and run `lerna repair`:

[version 6 - working](https://github.com/vollmerr/lerna-nx-cache-issue/tree/main/v6)
[version 7 - broken](https://github.com/vollmerr/lerna-nx-cache-issue/tree/main/v7)

When the nx.json is removed it works fine.

### Failure Logs / Configuration

<details><summary>lerna.json</summary><p>

```json
{
  "$schema": "node_modules/lerna/schemas/lerna-schema.json",
  "version": "0.0.0"
}
```

</p></details>

<details><summary>nx.json</summary><p>

```json
{
  "$schema": "node_modules/nx/schemas/nx-schema.json",
  "tasksRunnerOptions": {
    "default": {
      "runner": "nx/tasks-runners/default",
      "options": {
        "cacheableOperations": [
          "build",
          "test"
        ]
      }
    }
  },
  "namedInputs": {
    "default": [
      "{projectRoot}/**/*"
    ],
    "prod": [
      "!{projectRoot}/**/*.spec.tsx"
    ],
    "production": [
      "default"
    ]
  },
  "targetDefaults": {
    "build": {
      "inputs": [
        "prod",
        "^prod"
      ],
      "dependsOn": [
        "^build"
      ]
    },
    "test": {
      "inputs": [
        "default",
        "^prod"
      ]
    },
    "dev": {
      "dependsOn": [
        "^build"
      ]
    }
  }
}
```

</p></details>

### Environment

<details><summary>Environment info</summary><p>

```
  System:
    OS: Windows 10 10.0.22621
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i9-12900HK
  Binaries:
    Node: 18.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 3.5.1 - C:\Program Files\nodejs\yarn.CMD
    npm: 9.5.1 - C:\Program Files\nodejs\npm.CMD
  Utilities:
    Git: 2.40.1.
  npmPackages:
    lerna: ^7.0.0 => 7.0.0
```

</p></details>

**Comment by matteobruni** [comment]
Same for me, [This repository](https://github.com/matteobruni/tsparticles) has the nx.json file and it's broken, other repositories, [like this](https://github.com/tsparticles/presets), without the nx.json file are working.

**Comment by JamesHenry** [comment]
Thank you very much for the reproduction @vollmerr we will take a look and figure out a resolution

**Comment by JamesHenry** [comment]
Thanks again for the clear repro, this is fixed in lerna `7.0.1`
